### PR TITLE
explaining merging and error situation  

### PR DESCRIPTION
**Pull request by majidaldo**
#112

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -146,10 +146,10 @@ It is possible to define environment variables that should be configured in the
       DB_LOCATION: https://localhost/dev
 
 Environment variables defined in *list form* (like ``PATH`` and ``PYTHONPATH`` above) will **append** to existing
-variables, using the appropriate separator for the platform (``:`` on Linux/OSX and ``;`` on Windows).
+variables in a file using the appropriate separator for the platform (``:`` on Linux/OSX and ``;`` on Windows).
```
Comment (prusse-martin) [inline docs/usage.rst:149]:
```suggestion
environment variables with the values found in the ``.devenv.yml`` file, using the appropriate separator for the platform (``:`` on Linux/OSX and ``;`` on Windows).
```

File: docs/usage.rst [hunk 2]
```diff
@@ -146,10 +146,10 @@ It is possible to define environment variables that should be configured in the
...
 
 Environment variables defined as a single string (like ``DB_LOCATION`` above) will **overwrite** an existing
-variable with the same name.
+variable with the same name in a file.
```
Comment (prusse-martin) [inline docs/usage.rst:152]:
```suggestion
environment variable with the value from the ``.devenv.yml`` file.
```

File: docs/usage.rst [hunk 3]
```diff
@@ -200,9 +202,13 @@ In this setup, all the user has to do is executing ``conda devenv``:
     $ cd ~/projects/web-ui
     $ conda devenv
 
-This will create a ``conda`` environment named ``web-ui`` containing all the dependencies and environment variables
+This will create a ``conda`` environment named ``web-ui`` merging all the dependencies and environment variables
 defined in both files.
 
+However, the same environment variable defined as a single string (like DB_LOCATION above) in both files will raise an error
+unless it is not allowed to 'pass through' using the ``not is_included`` expression above as an example.
+In other words, an 'overwrite' situation is not allowed between files.
```
Comment (prusse-martin) [inline docs/usage.rst:210]:
```suggestion
However, the same environment variable defined as a single string (like ``DB_LOCATION`` above) in both files will raise an error
unless it is not allowed to 'pass through' using the ``# [not is_included]`` selector above as an example.
In other words, an 'overwrite' situation is not allowed between files.
```

The syntax `# [some_bool_expr]` have been copied from [conda build and they call it preprocessing selectors](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#preprocessing-selectors) I think we should you the same term. Actually [our docs describe the feature in jinja section](https://conda-devenv.readthedocs.io/en/latest/usage.html#jinja2) but fails to point the origin and I think we should use the same name and add an reference to the conda-build.

**Review by prusse-martin — CHANGES_REQUESTED** [review]

**Comment by prusse-martin** [comment]
I did add a link to conda-build's preprocessing selectors.

**Review by nicoddemus — APPROVED** [review]

**Comment by nicoddemus** [comment]
Thanks @majidaldo! 

@prusse-martin let's squash merge this. 👍

**Comment by prusse-martin** [comment]
Ty for the PR @majidaldo.

**Review by prusse-martin — APPROVED** [review]

**Pull request merged**